plugins/dns/plugin.go

```diff
@@ -37,20 +47,46 @@ func setupDNS4(args ...string) (handler.Handler4, error) {
 		if DNSServer.To4() == nil {
 			return Handler4, errors.New("expected an DNS server address, got: " + arg)
 		}
-		dnsServers = append(dnsServers, DNSServer)
+		dnsServers4 = append(dnsServers4, DNSServer)
 	}
-	log.Infof("loaded %d DNS servers.", len(dnsServers))
+	log.Infof("loaded %d DNS servers.", len(dnsServers4))
 	return Handler4, nil
 }
 
-// Handler6 not implemented only IPv4
+// Handler6 handles DHCPv6 packets for the dns plugin
 func Handler6(req, resp dhcpv6.DHCPv6) (dhcpv6.DHCPv6, bool) {
-	// TODO add DNS servers for v6 to the response
+	decap, err := req.GetInnerMessage()
+	if err != nil {
+		log.Errorf("Could not decapsulate relayed message, aborting: %v", err)
+		return nil, true
+	}
+
+	oro := decap.GetOneOption(dhcpv6.OptionORO)
+	if oro == nil {
+		// A client MUST include an Option Request option in a Solicit, Request,
+		// Renew, Rebind, or Information-request message to inform the server
+		// about options the client wants the server to send to the client
+		log.Warn("No OptionRequestOption in request, dropping")
+		return nil, true
+	}
+
+	for _, opt := range oro.(*dhcpv6.OptRequestedOption).RequestedOptions() {
```

Comment:
NIT: this can be replaced with the `IsOptionRequested` method:
https://github.com/insomniacslk/dhcp/blob/master/dhcpv6/dhcpv6message.go#L239

Reply:
Same as below, this doesn't differentiate "Option is there but doesn't contain the code" and "Option is not there". For v6 this matters less though, since clients MUST include the option; I guess we could act the same in both cases, but I think we should drop malformed packets as much as possible to avoid bad surprises later

Reply:
> but I think we should drop malformed packets as much as possible to avoid bad surprises later

I agree but this doesn't seem to me the proper place where to do packet validation. Should we move this validation somewhere in the library? Maybe in `NewRequestFromAdvertise` and `NewReplyFromMessage`.

Reply:
Right, it's probably a good idea to put validation in the library indeed.
I think it's better not to overload simple functions like NewYFromX which can be used to create useful synthetic messages for testing even if they're invalid, but we can discuss that in an issue in the library

I'll remove this check here and move to IsOptionRequested